libretroshare/src/rsserver/rsinit.cc

```diff
@@ -391,6 +392,22 @@ int RsInit::InitRetroShare(const RsConfigOptions& conf)
 	if(!RsAccounts::init(rsInitConfig->optBaseDir,error_code))
 		return error_code ;
 
+#ifdef RS_JSONAPI
+    // We create the JsonApiServer this early, because it is needed *before* login
+
+    RsInfo() << "Allocating jsonAPI server (not launched yet) " << std::endl;
+    JsonApiServer *jas = new JsonApiServer();
+    jas->setListeningPort(conf.jsonApiPort);
+    jas->setBindAddress(conf.jsonApiBindAddress);
```

Comment:
Is it still possible to set json api port and binding address from command line options?

Reply:
yes I think so. It's the responsibility of main() to set the port and bind address parameters, so that rsinit sees them in conf.